src/TensorFlowNET.Core/Eager/c_api.eager.cs

```diff
@@ -90,6 +91,35 @@ public partial class c_api
         [DllImport(TensorFlowLibName)]
         public static extern void TFE_DeleteContext(IntPtr ctx);
 
+        /// <summary>
+        /// Execute the operation defined by <paramref name="op"/> and return handles to computed
+        /// tensors in <paramref name="retvals"/>.
+        /// </summary>
+        /// <remarks>
+        /// Upon successful return, the first <paramref name="num_retvals"/> slots in <paramref name="retvals"/> will
+        /// contain handle instances which the caller is responsible for disposing once they are no longer in use.
+        /// </remarks>
+        /// <param name="op"></param>
+        /// <param name="retvals"></param>
+        /// <param name="num_retvals"></param>
+        /// <param name="status"></param>
+        public static void TFE_Execute(SafeOpHandle op, SafeTensorHandleHandle[] retvals, out int num_retvals, SafeStatusHandle status)
```

Comment:
📝 This is really the most "unpleasant" part of this change. The difficulty stems from the fact that `retvals` is created by the caller, and there is no trivial way to dispose of all the resulting handles. My recommendation is to keep this form for now, and focus on the higher-level API if you are interested in cleaning this up. I made the mistake of trying to "fix" this too quickly in an interop library I created in the past, and the result turned out to be more complex.